ui/src/common/common-slice.ts

```diff
@@ -38,12 +38,14 @@ export const initialChangeTimes: ChangeTimes = {
 export type CommonState = {
     version: string | undefined;
     changeTimes: ChangeTimes;
+    availableRoles: Role[];
     userPrivileges: Privilege[];
```

Comment:
Pitäisikö reduxiin tallentaa koko käyttäjä osittaisten tietojen sijaan?

Reply:
Juup, dailyjutskailujen jälkeen sen nykyisen aktiviisen käyttäjäroolinkin tallennuksen tarvitsee näyttääkseen siellä valintamenussa sen aktiivisen roolivalinnan, joten näkisin että olisi jo järkevää tallentaa se kokonainen User-rakenne reduksiin. Näin siis teinkin, ks committi. 

Sinänsähän tästä puhuttiin esikatselurefakin yhteydessä, että yleisenä tavoitteena olisi pitää reduksissa minimaalinen tilatieto, mutta tuntuu jotenkin oudolta tehdä jokin StoredUser = ... johon asettaa vaan ne tietyt tarpeelliset kentät (sehän User sisältää muutakin ns. tarpeetonta kuitenkin). 

Tuntuu kyllä omasta mielestä edelleen jotenkin oudolta reduksin tilaoptimoinnilta ylläpitää siellä osittaisia tietorakenteita, joista pitää sitten usein komponenteissa päätellä varsinaisten tietorakenteiden kanssa se oikeasti näytettävä tila hetkellisen datan kanssa (ei toki tämän Userin kanssa). Vertauksena siis edelleen se, että ne varsinaiset tietorakenteet voisivat vaan majailla tilatietonsa kanssa jo siellä Reduksissa kokonaisina, josta seuraisi omalla intuitiolla Redux-tilan helppokäyttäisyys sekä vähemmän koodi-/tyyppi-/tilapäivityskohinaa yms noista yhdistelyistä ja eri tietorakenteiden käytöistä tallennetussa tilassa vs oikeasti piirtävässä komponentissa. Tässä lienee tosiaan ollut se jokin probleema aiemmin, mitä en oikein kyllä vieläkään nää vaikka tästä jutskattiinkin, hmmh.